### PR TITLE
Rename save game variable for clarity

### LINKED ISSUE #9033: Change variable names on Saving games code example
**Your Godot version:**
4.2
**Issue description:**
The current code of the Saving games entry is
```gdscript
func save_game():
	var save_game = FileAccess.open("user://savegame.save", FileAccess.WRITE)
	var save_nodes = get_tree().get_nodes_in_group("Persist")
	for node in save_nodes:
		# Check the node is an instanced scene so it can be instanced again during load.
		if node.scene_file_path.is_empty():
			print("persistent node '%s' is not an instanced scene, skipped" % node.name)
			continue

		# Check the node has a save function.
		if !node.has_method("save"):
			print("persistent node '%s' is missing a save() function, skipped" % node.name)
			continue

		# Call the node's save function.
		var node_data = node.call("save")

		# JSON provides a static method to serialized JSON string.
		var json_string = JSON.stringify(node_data)

		# Store the save dictionary as a new line in the save file.
		save_game.store_line(json_string)

```

As it can be seen, the callable function/method `save_game` is named the same as the variable `save_game` which is confusing and isn't a good practice, and Godot may autocomplete it as `save_game()` in the last line causing recursion errors.

**URL to the documentation page:** https://docs.godotengine.org/en/stable/tutorials/io/saving_games.html

Possible fixes could be to rename the `save_game` variable to `save_data`, also avoiding to use the word "save" so many times could help on clarity (like using `game_data` and `game_data.save` for example).

### PR DESCRIPTION
**Pull request by skyace65**
save game is also the name of the function so I renamed it for clarity. Closes #9033.

### CODE REVIEW

File: tutorials/io/saving_games.rst [hunk 1]
```diff
@@ -141,7 +141,7 @@ way to pull the data out of the file as well.
     # Go through everything in the persist category and ask them to return a
     # dict of relevant variables.
     func save_game():
-        var save_game = FileAccess.open("user://savegame.save", FileAccess.WRITE)
+        var save_data = FileAccess.open("user://savegame.save", FileAccess.WRITE)
```
Comment (AThousandShips) [inline tutorials/io/saving_games.rst:144]:
```suggestion
        var save_file = FileAccess.open("user://savegame.save", FileAccess.WRITE)
```
I'd say this is clearer

**Comment by skyace65** [comment]
Updated with AThousandShips' suggestion.

**Review by AThousandShips — APPROVED** [review]

**Review by Piralein — APPROVED** [review]

**Review by mhilbrunner — APPROVED** [review]

**Comment by mhilbrunner** [comment]
Triple approval! Merged. :)

**Comment by mhilbrunner** [comment]
Cherry-picked to 4.2 in https://github.com/godotengine/godot-docs/pull/9647.

**Pull request merged**